### PR TITLE
Allow initialization with access credentials

### PR DESCRIPTION
**Pull request by avnerbraverman**
This is required for Outform

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -96,6 +96,23 @@ URL of the server hosting the Reshuffle runtime, and `RESHUFFLE_ENCRYPTION_KEY`
 which is a 64 digit hex string (32 bytes) which is used to encrypt the access
 tokens received from Salesforce during the authentication process.
 
+During the authntication process, Salesforce sends back credentials including
```
Comment (foundingnimo) [inline README.md]: authentication

File: README.md [hunk 2]
```diff
@@ -96,6 +96,23 @@ URL of the server hosting the Reshuffle runtime, and `RESHUFFLE_ENCRYPTION_KEY`
...
+and access token, a refresh token and an instance URL. If you already have
```
Comment (foundingnimo) [inline README.md]: an access token

File: README.md [hunk 3]
```diff
@@ -163,6 +180,10 @@ When Reshuffle is running on a server, you can use the `authenticator` utility
 provided with this connector to start the authentication process from your
 (or your user's) machine.
 
+This action throws en error if the connector already has Salesforce access
```
Comment (foundingnimo) [inline README.md]: an error

File: src/common.ts [hunk 4]
```diff
@@ -20,6 +20,17 @@ export function openAuthenticationWindow(
...
+  if (typeof url !== 'string') {
+    throw new Error(`Invalid url: ${url}`)
+  }
+  const match = url.match(/^(https:\/\/[\w-]+(\.[\w-]+)*(:\d{1,5})?)\/?$/)
+  if (!match) {
+    throw new Error(`Invalid url: ${url}`)
+  }
+  return match[1]
+}
+
```
Comment (foundingnimo) [inline src/common.ts:33]: I think the `validateUrl` function in `BaseConnector` can be used here

File: src/index.ts [hunk 5]
```diff
@@ -12,11 +12,12 @@ import {
 import {
   AUTHPATH,
   openAuthenticationWindow,
+  validateBaseURL,
```
Comment (foundingnimo) [inline src/index.ts:15]: Can be used from `BaseConnector`?

File: src/index.ts [hunk 6]
```diff
@@ -92,18 +110,15 @@ export class SalesforceConnector extends CoreConnector {
...
-        userId: userInfo.id,
-        organizationId: userInfo.organizationId,
-        userUrl: userInfo.url,
       }
       console.log(`SalesforceConnector: Authorized as ${
-        credentials.userId} of ${credentials.organizationId}`)
+        userInfo.id} of ${userInfo.organizationId} (${userInfo.url})`)
 
-      const enc = encrypt(JSON.stringify(credentials), this.key)
+      const enc = encrypt(JSON.stringify(this.credentials), this.key)
```
Comment (foundingnimo) [inline src/index.ts]: Suggest to use `encrypted` instead of `enc` - like it is a few lines above

File: src/index.ts (hunk 5 above)
Comment (avnerbraverman) [reply src/index.ts:15]: My BaseConnector doesn't seem to have this one. Maybe it's an older version but I don't want to update without testing again

File: src/common.ts (hunk 4 above)
Comment (avnerbraverman) [reply src/common.ts:33]: See below

File: README.md (hunk 3 above)
Comment (avnerbraverman) [reply README.md]: Fixed (this is eld English I wes oozing)

File: README.md (hunk 2 above)
Comment (avnerbraverman) [reply README.md]: fixed

File: README.md (hunk 1 above)
Comment (avnerbraverman) [reply README.md]: fixed

**Comment by avnerbraverman** [comment]
Thanks @duckranger !!

**Pull request merged**